### PR TITLE
Feature upload profile picture to aws s3 #8

### PR DESCRIPTION
**Pull request by Tojonirina4**
You can test the dialog in the userProfile [/profile] page

### CODE REVIEW

File: client/src/component/Uploader/FileUploaderDialog.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import DialogContent from "@material-ui/core/DialogContent";
+import DialogTitle from "@material-ui/core/DialogTitle";
+import Dropzone from "react-dropzone";
+import CircularProgress from '@material-ui/core/CircularProgress';
+import Button from "@material-ui/core/Button";
+
+
+const FileUploaderDialog = (props) => {
+  return (
+    <React.Fragment>
```
Comment (johnsciutto) [inline client/src/component/Uploader/FileUploaderDialog.js]: Personal preference here, but the new syntax for `Fragment` is a lot less verbose: `<></>` 😛

File: client/src/component/Uploader/FileUploaderDialog.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                  <input {...getInputProps()} />
+                  {props.file ? (
+                    <p>{props.file.name}</p>
+                  ) : (
+                    <p>
+                      Drag 'n' drop some files here, or click to select files
+                    </p>
+                  )}
+                </div>
+              </section>
```
Comment (johnsciutto) [inline client/src/component/Uploader/FileUploaderDialog.js:52]: I never used `react-dropzone`, but I take it based on this section that it doesn't play well with Material UI?

**Review by johnsciutto — COMMENTED** [review]
Nice work 👍🏻

File: client/src/pages/User/UserProfile.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    const formData = new FormData();
+    formData.append("file", selectedFile);
+    fetch("/api/file-upload", {
+      method: "POST",
+      body: formData,
+    })
+      .then((response) => response.json())
+      .then((result) => {
+        setDisplaySnack(true);
+        setDisplaySnackError(false);
```
Comment (sina-jamshidi) [inline client/src/pages/User/UserProfile.js]: these could be combined into one function that takes in whether it's an error or not the and the message to display. So we only need one Snackbar

File: client/src/pages/User/styles.css [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+.Dialog-container{
+    height: "50%";
+}
+
+.Progress{
+    color: #48ff00;
+    position: absolute;
+    top: 50%;
+    left: 50%;
+}
```
Comment (sina-jamshidi) [inline client/src/pages/User/styles.css]: we are going to use `makeStyles` for styling instead of CSS

File: server/routes/file-upload.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  const user = res.locals.user;
+  singleUpload(req, res, function(err) {
+    if (err) {
+      return res.status(422).send({errors: [{title: 'File Upload Error', detail: err.message}] });
+    }
+    user.pictureUrl = req.file.location;
+    user.save();
+    return res.json({'imageUrl': req.file.location});
+  });
+});
```
Comment (sina-jamshidi) [inline server/routes/file-upload.js:18]: 👍

**Review by sina-jamshidi — APPROVED** [review]

File: client/src/pages/User/UserProfile.js (hunk 3 above)
Comment (Tojonirina4) [reply client/src/pages/User/UserProfile.js:45]: good point. thx

File: client/src/component/Uploader/FileUploaderDialog.js (hunk 2 above)
Comment (Tojonirina4) [reply client/src/component/Uploader/FileUploaderDialog.js:52]: Hello John, I'm not sure if I understand your question. Could you please elaborate?

File: client/src/pages/User/styles.css (hunk 4 above)
Comment (Tojonirina4) [reply client/src/pages/User/styles.css]: great news :)

**Pull request merged**